src/display/api.js

```diff
@@ -1256,7 +1272,7 @@ class PDFPageProxy {
    *   children, very similar to a HTML DOM tree), or `null` if no XFA exists.
    */
   getXfa() {
-    return (this._xfaPromise ||= this._transport.getPageXfa(this._pageIndex));
+    return (this._xfaPromise ||= Promise.resolve(this._xfaHtml));
```

Comment:
With this being *synchronously* available, I'm not sure that caching a Promise is still necessary!?
Can't you simply remove `this._xfaPromise` completely and instead do e.g.
```javascript
async getXFA() {
  return this._transport._htmlForXfa;
```